akka-actor/src/main/scala/akka/io/SimpleDnsCache.scala

```diff
@@ -64,8 +65,8 @@ object SimpleDnsCache {
       } yield e.answer
     }
 
-    def put(name: K, answer: V, ttlMillis: Long): Cache[K, V] = {
-      val until0 = clock() + ttlMillis
+    def put(name: K, answer: V, ttl: Duration): Cache[K, V] = {
+      val until0 = clock() + ttl.toMillis
```

Comment:
This will throw `IllegalArgumentException` if `Duration` isn't a `FiniteDuration`, e.g. `Inf`

I think we should use `ttl: FiniteDuration` here.